### PR TITLE
Fixed xcube plugin auto-recognition in case a plugin project uses `pyproject.toml` file.

### LINKED ISSUE #963: xcube plugins not auto-recognized when using `pyproject.toml`
**Describe the bug**

xcube plugin packages are not automatically recognized as xcube plugins 

1. although the package name is prefixed using `xcube_`;
2. if installed via `pip -e .`;
3. if using a `pyproject.toml` rather than `setup.py`.

**To Reproduce**

See plugin [xcube-cmems](https://github.com/xcube-dev/xcube-cmems) that is forced to explicitly declare the entry point `xcube_plugins` in its `pyproject.toml`. 

**Expected behavior**

Packages named `xcube_<name>` should automatically be recognized as xcube plugins.

### PR DESCRIPTION
**Pull request by konstntokas**
Closes #963

Checklist:

* [ ] ~Add unit tests and/or doctests in docstrings~
* [ ] ~Add docstrings and API docs for any new/modified user-facing classes and functions~
* [ ] ~New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [ ] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: xcube/util/plugin.py [hunk 1]
```diff
@@ -81,6 +82,13 @@ def discover_plugin_modules(module_prefixes=None):
             #  but logging is not yet configured at this point.
             # print(f'xcube plugin module found: {module_name}')
             entry_points.append(_ModuleEntryPoint(module_name))
+
+    # discover plugins if they are only installed in a editable mode
+    for module_prefix in module_prefixes:
+        for path_hook in fnmatch.filter(sys.path, f'__editable__.{module_prefix}*.finder.__path_hook__'):
+            module_name = path_hook.split("-")[0].split(".")[1]
+            entry_points.append(_ModuleEntryPoint(module_name))
```
[the remark this answers is not in the document]
Comment (forman) [reply xcube/util/plugin.py:90]: Could you please explain and/or provide a reference why and when `sys.path` contains `...__path_hook__`.

**Review by forman — CHANGES_REQUESTED** [review]

**Comment by forman** [comment]
Also, please provide a clear human-readable title for the PR and also always provide a PR description, at least: "Closes #963".

File: xcube/util/plugin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (konstntokas) [reply xcube/util/plugin.py]: I documented it in #963.

File: xcube/util/plugin.py [hunk 2]
```diff
@@ -64,6 +65,16 @@ def discover_plugin_modules(module_prefixes=None):
             #  but logging is not yet configured at this point.
             # print(f'xcube plugin module found: {module_name}')
             entry_points.append(_ModuleEntryPoint(module_name))
+
+    # discover plugins if they are installed in a editable mode (see PR #963)
```
Comment (forman) [inline xcube/util/plugin.py:69]:
```suggestion
    # Discover plugins if they are installed in a editable mode.
    # See https://github.com/xcube-dev/xcube/issues/963
```

Comment (forman) [reply xcube/util/plugin.py:69]: Link to the _issue_, not the PR.

**Review by forman — CHANGES_REQUESTED** [review]
Please look at my suggestion. Note, PR descriptions should not be put in brackets. Just replace the whole text `[Description of PR]` by your description text.

**Comment by forman** [comment]
Btw, why are the following from the PR checklist not applicable?

- Add unit tests and/or doctests in docstrings
- Add docstrings and API docs for any new/modified user-facing classes and functions
- New/modified features documented in docs/source/*

**Comment by forman** [comment]
Removed @TejasMorbagal as he is extremely busy.

File: xcube/util/plugin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (forman) [reply xcube/util/plugin.py]: > I documented it in #963.

Thanks!

**Review by forman — APPROVED** [review]

**Pull request merged**